lib/ScssCompiler.js

```diff
@@ -0,0 +1,280 @@
+const _ = require('lodash');
+const path = require('path');
+const nodeSassFork = require('@bigcommerce/node-sass');
+const dartSass = require('sass');
+const Fiber = require('fibers');
+const { promisify } = require('util');
+
+class ScssCompiler {
+    /**
+     * WARNING! Make sure to create a new instance per each compilation to avoid processes interfering each other
+     *
+     * @param {object} logger
+     * @param {object} engines
+     * @param {object} engines.primary
+     * @param {object} engines.fallback
+     * @param {function} logger.error
+     */
+    constructor(
+        logger,
+        engines = {
+            primary: dartSass,
+            fallback: nodeSassFork,
+        },
+    ) {
+        this.logger = logger;
+        this.engines = engines;
+        this.files = {};
+        this.fullUrls = {};
+
+        this.activatePrimaryEngine();
+    }
+
+    activatePrimaryEngine() {
+        this.engines.active = this.engines.primary;
+    }
+
+    activateFallbackEngine() {
+        this.engines.active = this.engines.fallback;
+    }
+
+    /**
+     * Compile SCSS into CSS and return the content
+     *
+     * @public
+     * @param options
+     */
+    async compile(options) {
+        this.files = options.files || {};
+        const engineOptions = {
+            data: options.data,
+            outFile: options.dest,
+            files: options.files,
+            sourceMap: options.sourceMap,
+            sourceMapEmbed: options.sourceMap,
+            fiber: Fiber,
+            functions: this.getScssFunctions(options.themeSettings),
+            importer: this.scssImporter.bind(this),
+        };
+        let result = {};
+        this.activatePrimaryEngine();
+        try {
+            result = await this.runEngine(engineOptions);
+        } catch (dartSassErr) {
+            this.logger.error(`Error during css compilation by primary engine:\n${dartSassErr}\nWill retry with a fallback engine`);
+            this.activateFallbackEngine();
+            result = await this.runEngine(engineOptions);
+        }
+
+        return result.css;
+    }
+
+    /**
+     * Calls active engine with the specified options
+     *
+     * @private
+     * @param options
+     * @return {Promise<object>}
+     */
+    async runEngine(options) {
```

Comment:
should we just call this `render` ? 🍹

Reply:
Ok, I can rename it to render.
I named it `runEngine ` because it runs the active engine for real 🙂
Also, having 2 methods "render" and "compile" can be confusing a bit